go-controller/pkg/ovn/controller/apbroute/gateway_info/gateway_info.go

```diff
@@ -148,9 +148,9 @@ func (g *GatewayInfoList) Equal(g2 *GatewayInfoList) bool {
 }
 
 type GatewayInfo struct {
-	Gateways   sets.Set[string]
-	BFDEnabled bool
-	applied    bool
+	Gateways      sets.Set[string]
+	BFDEnabled    bool
+	failedToApply bool
```

Comment:
Renamed to `failedToApply` to align with `false` as default since it follows the golang idiomatic behavior for booleans, rather than init to `true` and change to `false` when it fails to apply.